packages/bridge-controller/src/selectors.ts

```diff
@@ -0,0 +1,305 @@
+import type {
+  CurrencyRateState,
+  MultichainAssetsRatesControllerState,
+  TokenRatesControllerState,
+} from '@metamask/assets-controllers';
+import type { CaipAssetType } from '@metamask/utils';
+import { isStrictHexString } from '@metamask/utils';
+import { orderBy } from 'lodash';
+import { createSelector } from 'reselect';
+
+import type {
+  BridgeControllerState,
+  BridgeFeatureFlagsKey,
+  GenericQuoteRequest,
+  QuoteMetadata,
+  QuoteResponse,
+  SolanaFees,
+  TokenAmountValues,
+} from './types';
+import { RequestStatus, SortOrder } from './types';
+import {
+  getNativeAssetForChainId,
+  isNativeAddress,
+  isSolanaChainId,
+} from './utils/bridge';
+import {
+  formatAddressToAssetId,
+  formatChainIdToHex,
+} from './utils/caip-formatters';
+import {
+  calcRelayerFee,
+  calcSentAmount,
+  calcSwapRate,
+  calcToAmount,
+  getQuoteIdentifier,
+} from './utils/quote';
+
+/**
+ * The controller states that provide exchange rates
+ */
+type ExchangeRateControllers = MultichainAssetsRatesControllerState &
```

Comment:
Nit: Might want to rename this for better readability

```suggestion
type ExchangeRateControllersState = MultichainAssetsRatesControllerState &
```